### PR TITLE
Rdma ns aware support

### PR DESCRIPTION
**Pull request by adrianchiris**
This PR adds support for network namespace isolation for RDMA devices.

This PR is required as current kernel model is to recreate IPoIB netdevices (it actually recreates all ULP resources associated with it causing the IPoIB netdevice to recreate) in default namespace when RDMA device is moved to a namespace.

hence in contrast to Ethernet case where one would use [SR-IOV CNI](https://github.com/intel/sriov-cni) and [RDMA CNI](https://github.com/Mellanox/rdma-cni) in a chain to achieve this. Here we need to surgically add support.

If/When a better model is introduced in kernel, the code may be removed.

### CODE REVIEW

File: glide.lock [hunk 1]
```diff
@@ -28,6 +28,13 @@ imports:
   version: d8f796af33cc11cb798c1aaeb27a4ebc5099927d
   subpackages:
   - spew
+- name: github.com/Mellanox/rdma-cni
+  version: 4066382292e068c67d51dbf6726a94cda1193662
+  subpackages:
+  - pkg/rdma
+  - pkg/types
+- name: github.com/Mellanox/rdmamap
+  version: 0200e5fda04d08287707ee29fcd73f5b51d59dfa
```
Comment (moshe010) [inline glide.lock]: let take the commit with the go mod change

File: README.md [hunk 2]
```diff
@@ -196,8 +196,9 @@ echo 8 > /sys/class/net/ib0/device/sriov_numvfs
 * `guid` (string, optional): InfiniBand Guid for VF.
 * `pkey` (string, optional): InfiniBand pkey for VF, this filed is used by [ib-kubernetes](https://www.github.com/Mellanox/ib-kubernetes) to add pkey with guid to InfiniBand subnet manager client e.g. [Mellanox UFM](https://www.mellanox.com/products/management-software/ufm), [OpenSM](https://docs.mellanox.com/display/MLNXOFEDv461000/OpenSM).
 * `ipam` (dictionary, optional): IPAM configuration to be used for this network, `dhcp` is not supported.
-* `link_state` (dictionary, optional): Enforces link state for the VF. Allowed values: auto, enable, disable.
-
+* `link_state` (string, optional): Enforces link state for the VF. Allowed values: auto, enable, disable.
+* `rdmaIsolation` (boolean, optional): Enable RDMA network namespace isolation for RDMA workloads. More information
```
Comment (moshe010) [inline README.md]: maybe add a remake that you shouldn't use rdma-cni if rdmaIsolation is true

**Comment by adrianchiris** [comment]
> Did Mamduh test it?
He started to. pod is created /destroyed as expected.

**Comment by Mmduh-483** [comment]
> > Did Mamduh test it?
> > He started to. pod is created /destroyed as expected.

@moshe010 I sent the full testing report this morning, @adrianchiris check issue 3 it is critical

**Pull request merged**